### ISSUE TITLE
Signature different when using `--forward-to` from localhost Docker

### ISSUE THREAD

**Issue by mrowles** [stripe/stripe-cli#817]
I've set up Stripe CLI via Docker which forwards to a local Firebase function. I've followed all of the paths but seems to me like it's a setting I'm missing or a potential bug in the latest version.

I have a suspicion it's because I'm using `docker --forward-to` parameter and I'm meant to be doing an additional step to make the rawBody correct, perhaps a URL parameter is wrong or something due to the forward (?)

### Issue
When trying to construct the event locally (`constructEvent` function), I get the error:

> 'No signatures found matching the expected signature for payload. Are you passing the raw request body you received from Stripe? https://github.com/stripe/stripe-node#webhook-signing'

### Expected Behavior
I expect the event to be constructed. _This does not happen when I deploy, this is only on localhost - the webhook works fine when I deploy._

### Steps to reproduce

1. I set the webhook signing secret to the local env:

*stripe-setup.sh*
```
#!/bin/sh

STRIPE_API_KEY=$(firebase functions:config:get stripe.api_key | tr -d '"')
export STRIPE_API_KEY

STRIPE_WEBHOOK_SIGNING_SECRET=$(docker run --network="host" --rm -it stripe/stripe-cli:latest listen --print-secret --api-key "$STRIPE_API_KEY")
export STRIPE_WEBHOOK_SIGNING_SECRET
```

2. I start the webhook local docker handler + forwarding

*stripe-webhook.sh*
```
#!/bin/sh

docker run --network="host" --rm -it stripe/stripe-cli:latest listen \
  --api-key "$STRIPE_API_KEY" \
  --forward-to host.docker.internal:5001/product/region/stripeWebhookHandler
```

Note: both of these `docker run` statements output the same `whsec_` security signing key.

3. I attempt to construct and handle the event:

*stripeWebhookHandler.ts*
```
export async function stripeWebhookHandler(request, response) {
    try {
        const sig = request.headers['stripe-signature'];
        const payload = request.rawBody; // when using firebase functions, it is suggested to use rawBody
        const secret = process.env.STRIPE_WEBHOOK_SIGNING_SECRET; // this is the same value as the shell scripts

        if (!sig) {
            throw new Error('Stripe Webhook Error: no sig found');
        } else if (!payload) {
            throw new Error('Stripe Webhook Error: no payload found');
        } else if (!secret) {
            throw new Error('Stripe Webhook Error: no secret found');
        }

        // this is where the error occurs 👇

        const event = stripe.webhooks.constructEvent(
            payload,
            sig,
            secret,
        );

        await handleStripeEventMiddleware(event);

        response.status(200).send('Webhook Handled');
    } catch (err) {
        response.status(500).send('Could not process Stripe Webhook event');
    }
}
```

### Traceback
```
StripeSignatureVerificationError: No signatures found matching the expected signature for payload. Are you passing the raw request body you received from Stripe? https://github.com/stripe/stripe-node#webhook-signing
>      at validateComputedSignature (/Users/user/company/repo/node_modules/stripe/lib/Webhooks.js:208:11)
>      at Object.verifyHeader (/Users/user/company/repo/node_modules/stripe/lib/Webhooks.js:103:5)
>      at Object.constructEvent (/Users/user/company/repo/node_modules/stripe/lib/Webhooks.js:10:20)
>      at handler (/Users/user/company/repo/packages/functions/src/stripeWebhookHandler.ts:27:36)
```

### Environment
macOS

**Comment by mrowles** [comment]
I should also mention that rawBody seems to be the correct request attribute to use, since it works after deploy, but I have tried other options such as rawBody.toString() and just body + JSON.stringing(body), no such luck

**Comment by suz-stripe** [comment]
@mrowles hello and sorry for the slow response to your issue! I wanted to thank you for providing such clear steps to reproduce and for the details you included. 

I took a look at the way the environment variable is being set locally in step 1. I ran the value assigned to `STRIPE_WEBHOOK_SIGNING_SECRET` through a hex dump tool and saw a suspicious looking character at the end:

```bash
$ echo -n $STRIPE_WEBHOOK_SIGNING_SECRET | xxd
00000000: 7768 7365 635f 3337 3266 6265 6330 6131  whsec_372fbec0a1
[ snip...]
00000030: 6536 6266 6538 6666 6239 6562 6535 3633  e6bfe8ffb9ebe563
00000040: 6333 3563 6262 0d                        c35cbb.
```
That invisible trailing character `0d` is a carriage return, which would definitely cause the `STRIPE_WEBHOOK_SIGNING_SECRET` to fail when passed to the `stripe.webhooks.constructEvent` call. This is just speculation on my end, so are you able to try reproducing this in your local environment too?

If you're still looking to use this approach, I recommend piping the results of the docker call to `| tr -d '\r'` to remove the trailing newline / carriage return. So the final line would look like this:
`STRIPE_WEBHOOK_SIGNING_SECRET=$(docker run --network="host" --rm -it stripe/stripe-cli:latest listen --print-secret --api-key "$STRIPE_API_KEY" |  tr -d '\r')`

Let me know if that helps! 🙂
